### PR TITLE
Add kube version flag

### LINKED ISSUE #722: kubeVersion is being loaded from helmfile.yaml instead of cli
### Operating system

macOS Ventura

### Helmfile Version

v0.151.0 arm64

### Helm Version

v3.11.1

### Bug description

The current helmfile [official doc](https://helmfile.readthedocs.io/en/latest/) specify this:
`helmfile.yaml`
```
# DEPRECATED: This is available only on Helm 2, which has been EOL since 2020
# Configure a Kubernetes version to  pass to 'helm template' via the --kube-version flag:
# See https://github.com/roboll/helmfile/pull/2002 for more information.
kubeVersion: v1.21
```

I've been working with an issue trying to specify the kubeVersion via CLI as the doc said but it's not working.

I've found [this](https://github.com/helmfile/helmfile/blob/f629ee1ae6540dcb1b3c9917bf3dbc81f96a07c0/pkg/state/state.go#L1205) in the code so I believe that helmfile is getting the kubeVersion from the release defined in the `helmfile.yaml` file instead of CLI as doc says.

Is this expected? Am I missing anything?

Thanks in advance.

```
### Example helmfile.yaml

environments:
  <env name>:
    missingFileHandler: Warn
    values: # These values are the ones used to render values.yaml.gotmpl
      - values.yaml

templates:
  default: &default
    chart: .
    namespace: argo-cd-system
    missingFileHandler: Warn
    version: 1.0
    kubeVersion: 1.18.0
    set:
      - name: foo
        value: bar
      - name: foo1
        value: bar1
    values: # Values to render the helm chart
      - <n values files>
    jsonPatches:
      - target:
          group: apps
          version: v1
          kind: Deployment
          name: redis-ha-haproxy
        path: jsonpatches/patch.redis-ha-haproxy-deployment.yaml
      - target:
          group: apps
          version: v1
          kind: StatefulSet
          name: redis-ha-server
        path: jsonpatches/patch.redis-ha-statefulset.yaml
      - target:
          version: v1
          kind: Pod
          name: redis-ha-configmap-test
        path: jsonpatches/patch.test-redis-ha-configmap.yaml
      - target:
          version: v1
          kind: Pod
          name: redis-ha-service-test
        path: jsonpatches/patch.test-redis-ha-pod.yaml

releases:
  - name: kraken
    <<: *default
```

### Error message you've seen (if any)

I'm working with the argocd helm chart v5.8.7.

So, if I render the templates specifying the kubeVersion in the `helmfile.yaml` file, I get the expected output:
```
apiVersion: autoscaling/v2beta1
```

But if I remove the kubeVersion from the `helmfile.yaml` I get the wrong output:
```
apiVersion: autoscaling/v2 #this apiVersion is not available in my k8s version`
```

### Steps to reproduce

N/A

### Working Helmfile Version

v0.151.0

### Relevant discussion

_No response_

### PR DESCRIPTION
**Pull request by FeLvi-zzz**
Add `--kube-version` flag into `helmfile template`.
It overrides `kubeVersion` in helmfile.yaml.

resolve #722

### CODE REVIEW

File: cmd/template.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ func NewTemplateCmd(globalCfg *config.GlobalImpl) *cobra.Command {
 	f.BoolVar(&templateOptions.IncludeTransitiveNeeds, "include-transitive-needs", false, `like --include-needs, but also includes transitive needs (needs of needs). Does nothing when --selector/-l flag is not provided. Overrides exclusions of other selectors and conditions.`)
 	f.BoolVar(&templateOptions.SkipCleanup, "skip-cleanup", false, "Stop cleaning up temporary values generated by helmfile and helm-secrets. Useful for debugging. Don't use in production for security")
 	f.StringVar(&templateOptions.PostRenderer, "post-renderer", "", `pass --post-renderer to "helm template" or "helm upgrade --install"`)
+	f.StringVar(&templateOptions.KubeVersion, "kube-version", "", `pass --kube-version to "helm template". Overrides kubeVersion in a yaml file`)
```
Comment (yxxhero) [inline cmd/template.go:48]:
```suggestion
	f.StringVar(&templateOptions.KubeVersion, "kube-version", "", `pass --kube-version to "helm template". Overrides kubeVersion in helmfile.yaml`)
```

File: pkg/state/state.go [hunk 2]
```diff
@@ -2625,7 +2628,9 @@ func (st *HelmState) appendApiVersionsFlags(flags []string, r *ReleaseSpec) []st
 		}
 	}
 
-	if r.KubeVersion != "" {
+	if kubeVersion != "" {
```
Comment (yxxhero) [inline pkg/state/state.go:2631]: try to use `switch`.

**Comment by yxxhero** [comment]
@FeLvi-zzz did only `template` subcommand need this flag?

**Comment by FeLvi-zzz** [comment]
yes, I suppose that other subcommands do not need `--kube-version` flag because they connect a k8s cluster and be able to get its version.

**Comment by mumoshu** [comment]
Hey @FeLvi-zzz, you made a good point!

It's true most of helmfile and helm commands don't take `--kube-version` due to the reason you mentioned.

However, not only `helm template` but also `helm-diff` takes another similar flag `--api-versions` that can potentially be obtained from the target cluster. I think that's why we already call `appendApiVersionsFlags` in the helmfile-diff code.

HOWEVER, I realized that helm-diff lacks support for `--kube-version`! So it turns out there's an existing bug that when you set `kubeVersion` in helmfile.yaml, helm-diff run via helmfile-diff would fail due to an invalid flag error. We need to enhance helm-diff for that later.

Until then, the way you did it in the helmfile-diff code, `flags = st.appendApiVersionsFlags(flags, release, "")`, looks correct!

**Comment by yxxhero** [comment]
> Hey @FeLvi-zzz, you made a good point!
> 
> It's true most of helmfile and helm commands don't take `--kube-version` due to the reason you mentioned.
> 
> However, not only `helm template` but also `helm-diff` takes another similar flag `--api-versions` where corresponding values can potentially be obtained from the target cluster. I think that's why we already call `appendApiVersionsFlags` in the helmfile-diff code.
> 
> HOWEVER, I realized that helm-diff lacks support for `--kube-version`! So it turns out there's an existing bug that when you set `kubeVersion` in helmfile.yaml, helm-diff run via helmfile-diff would fail due to an invalid flag error. We need to enhance helm-diff for that later.
> 
> Until then, the way you did it in the helmfile-diff code, `flags = st.appendApiVersionsFlags(flags, release, "")`, looks correct!

@FeLvi-zzz @mumoshu  I think that we can add some TODO comments for `flags = st.appendApiVersionsFlags(flags, release, "")`.

**Comment by yxxhero** [comment]
@FeLvi-zzz please add required comments for `flags = st.appendApiVersionsFlags(flags, release, "")`.

**Comment by FeLvi-zzz** [comment]
@yxxhero 
added the comment!

**Review by yxxhero — APPROVED** [review]

**Pull request merged**